.github/workflows/lint.yaml

```diff
@@ -1,22 +1,18 @@
 name: lint
 
-on:
-  pull_request:
-    paths:
-      - '**'
 
+on:
   push:
     branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      - develop-1.9
       - develop
-    paths:
```

Comment:
There's a limitation on the number of simultaneous CI jobs for the organization so I don't think this should be removed, even if it's just a couple of minutes saved that makes a difference when you are waiting for something to finish.

(There's also an argument for not burning non-productive compute cycles for environmental reasons.)